core/src/test/java/org/pentaho/platform/dataaccess/datasource/api/MetadataServiceTest.java

```diff
@@ -678,5 +680,14 @@ public void createNewByteArrayInputStream() throws Exception {
     assertNotNull( stream );
   }
 
+  @Test
+  public void extractXmiFromZipTest() throws Exception {
+    File f = new File( "target/test-classes/import_metadata.zip" );
```

Comment:
Your code should be portable and not have hard coded paths to test files, especially to compilation directories such as "target/test-classes". The folder naming conventions could change and thus your test will break. 

Java handles loading class resources, so you should assume that maven or gradle will handle it. So now you only have to worry about grabbing the files from Java resources:

Pentaho examples:
https://github.com/pentaho/pentaho-kettle/blob/master/engine/src/test/java/org/pentaho/di/core/row/ValueDataUtilTest.java#L212

- will grab a file from: src/test/resources/org/pentaho/di/core/row/txt-sample.txt

Basic java overview:
https://mkyong.com/java/java-read-a-file-from-resources-folder/

- examples on how to grab files from: src/main/resources/json/file2.json